cores/esp32/HardwareSerial.cpp

```diff
@@ -341,13 +341,50 @@ void HardwareSerial::begin(unsigned long baud, uint32_t config, int8_t rxPin, in
       case UART_NUM_2:
         if (rxPin < 0 && txPin < 0) {
           // do not change RX2/TX2 if it has already been set before
+#ifdef RX2
           rxPin = _rxPin < 0 ? (int8_t)RX2 : _rxPin;
```

Comment:
what if _rxPin is >= 0? What will set the pins if RX2 is not defined?

Reply:
Should it be 
```cpp
#else
    rxPin = _rxPin;
#endif
```

Reply:
> what if _rxPin is >= 0? What will set the pins if RX2 is not defined?

pins will be kept as -1 and if those are both -1, begin() will issue a log_e() and return doing nothing.

Reply:
_rxPin _txPin are the previously set pins, so yes, it should be used.
Otherwise it will -1, as the default function parameters.
Done.